### PR TITLE
update liberty wssecurity and saml sso prereqs

### PR DESCRIPTION
**Pull request by arunavemulapalli**
Making prereq updates required to update wsSecurity, wsSecuritySaml and samlWeb features and runtime to work with latest levels of cxf/ws-security (3.3.x) , wss4j (2.3.0) and opensaml(3.4.5).

### CODE REVIEW

**Comment by arunavemulapalli** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.wssecurity_fat.cxf,com.ibm.ws.security.saml.sso_fat

**Comment by arunavemulapalli** [comment]
#build

**Comment by andrea-pichler** [comment]
No L2 message review needed for translations

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
#run-libby-bot

File: dev/com.ibm.ws.org.ehcache.ehcache-107.3.8.1/bnd.overrides [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ javax.xml.parsers,\
+ org.osgi.framework, \
+ org.slf4j,\
+ org.xml.sax,\
+ org.xml.sax.helpers
+   
+Include-Resource: \
+ @${repo;org.ehcache.modules:ehcache-107;3.8.1}!/!META-INF/MANIFEST.MF|META-INF/maven/*
+
+Archiver-Version:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.ehcache.ehcache-107.3.8.1/bnd.overrides:45]: Are these necessary?

File: dev/com.ibm.ws.org.cryptacular.cryptacular/build.gradle [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.cryptacular.cryptacular/build.gradle:2]:
```suggestion
 * Copyright (c) 2020 IBM Corporation and others.
```

File: dev/com.ibm.ws.org.cryptacular.cryptacular/bnd.overrides [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Bundle-SymbolicName: com.ibm.ws.org.cryptacular.cryptacular.1.2.4
+Bundle-ActivationPolicy: lazy
+
+Import-Package: \
+ !org.bouncycastle.*,\
+ !org.bouncycastle.asn1,\
+ !org.bouncycastle.asn1.x500,\
+ !org.bouncycastle.asn1.x509
+
+Archiver-Version:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.cryptacular.cryptacular/bnd.overrides:25]: Are these necessary?

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-web.2.3.0/bnd.overrides [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+Export-Package: \
+ org.apache.wss4j.*           
+ 
+Import-Package: \
+ javax.servlet
+ 
+Include-Resource: \
+ @${repo;org.apache.wss4j:wss4j-ws-security-web;2.3.0}!/!META-INF/MANIFEST.MF|META-INF/maven/*
+
+Build-Jdk:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-web.2.3.0/bnd.overrides:30]: Are these necessary?

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-web.2.3.0/bnd.bnd [hunk 5]
```diff
@@ -0,0 +1,15 @@
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-web.2.3.0/bnd.bnd:2]:
```suggestion
# Copyright (c) 2020 IBM Corporation and others.
```

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-stax.2.3.0/bnd.overrides [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+WS-TraceGroup: WSS4J
+
+Export-Package: \
+ org.apache.wss4j.*           
+ 
+Include-Resource: \
+ @${repo;org.apache.wss4j:wss4j-ws-security-stax;2.3.0}!/!META-INF/MANIFEST.MF|META-INF/maven/*
+
+
+Build-Jdk:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-stax.2.3.0/bnd.overrides:28]: Are these necessary?

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-policy-stax.2.3.0/bnd.overrides [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+ javax.xml.bind.*, \
+ !com.ibm.security.jgss, \
+ !com.sun.security.jgss, \
+ !org.bouncycastle.asn1.*, \
+ * 
+Include-Resource: \
+ @${repo;org.apache.wss4j:wss4j-ws-security-policy-stax;2.3.0}!/!META-INF/**
+
+
+Build-Jdk:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-policy-stax.2.3.0/bnd.overrides:34]: Are these necessary?

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-dom.2.3.0/bnd.overrides [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+WS-TraceGroup: WSS4J
+
+Export-Package: \
+ org.apache.wss4j.*           
+ 
+Include-Resource: \
+ @${repo;org.apache.wss4j:wss4j-ws-security-dom;2.3.0}!/!META-INF/MANIFEST.MF|META-INF/maven/*
+
+
+Build-Jdk:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-dom.2.3.0/bnd.overrides:28]: Are these necessary?

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-common.2.3.0/bnd.overrides [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+Include-Resource: \
+ @${repo;org.apache.wss4j:wss4j-ws-security-common;2.3.0}!/!META-INF/MANIFEST.MF|META-INF/maven/*,\
+ @${repo;org.apache.wss4j:wss4j-ws-security-common;2.3.0}!/templates/**,\
+ @${repo;org.apache.wss4j:wss4j-ws-security-common;2.3.0}!/schemas/**,\
+ @${repo;org.apache.wss4j:wss4j-ws-security-common;2.3.0}!/*xml, \
+ @${repo;org.apache.wss4j:wss4j-ws-security-common;2.3.0}!/*xml, \
+ @${repo;org.bouncycastle:bcprov-jdk15on;1.61;EXACT}!/!META-INF/MANIFEST.MF|META-INF/maven/*
+
+
+Build-Jdk:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-common.2.3.0/bnd.overrides:47]: Are these necessary?

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-common.2.3.0/bnd.bnd [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-ws-security-common.2.3.0/bnd.bnd:2]:
```suggestion
# Copyright (c) 2020 IBM Corporation and others.
```

File: dev/com.ibm.ws.org.apache.wss4j-policy.2.3.0/bnd.overrides [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+Export-Package: \
+ org.apache.wss4j.*           
+ 
+ 
+Include-Resource: \
+ @${repo;org.apache.wss4j:wss4j-policy;2.3.0}!/!META-INF/MANIFEST.MF,\
+ @${repo;org.apache.wss4j:wss4j-policy;2.3.0}!/templates/**,\
+ @${repo;org.apache.wss4j:wss4j-policy;2.3.0}!/schemas/**,\
+ @${repo;org.apache.wss4j:wss4j-policy;2.3.0}!/*xml,\
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-policy.2.3.0/bnd.overrides:29]:
```suggestion
 @${repo;org.apache.wss4j:wss4j-policy;2.3.0}!/*xml
```

File: dev/com.ibm.ws.org.apache.wss4j-policy.2.3.0/bnd.overrides [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+
+Build-Jdk:
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.apache.wss4j-policy.2.3.0/bnd.overrides:31]: Are these necessary?

**Review by neuwerk — CHANGES_REQUESTED** [review]
Let me know when you have the renames updated. By and large these are minor changes, sorry for the shear number of them, but I wanted to make sure you didn't miss any.

File: dev/com.ibm.ws.net.shibboleth.utilities.java.support.7.5.1/bnd.overrides [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ 
+ Include-Resource: \
+  @${repo;net.shibboleth.utilities:java-support;7.5.1}!/!META-INF/MANIFEST.MF|META-INF/maven/*, \
+  @${repo;org.bouncycastle:bcprov-jdk15on;1.61;EXACT}, \
+  @${repo;org.bouncycastle:bcpkix-jdk15on;1.61;EXACT}
+
+Archiver-Version:
+Build-Jdk:
+Built-By:
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.net.shibboleth.utilities.java.support.7.5.1/bnd.overrides:55]: Probably not. these files are copied from the existing projects

File: dev/com.ibm.ws.org.apache.cxf-rt-security-saml.3.3/bnd.overrides [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+Built-By:
+Bundle-DocURL:
+Bundle-License:
+Implementation-Title:
+Implementation-Vendor:
+Implementation-Vendor-Id:
+Implementation-Version:
+Specification-Title:
+Specification-Vendor:
+Specification-Version:
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.cxf-rt-security-saml.3.3/bnd.overrides]: not sure. I can clean these up (if I don't see any use for these)when we move them to 3.4.1?

File: dev/com.ibm.ws.org.apache.cxf-rt-security-saml.3.3/build.gradle [hunk 15]
```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.cxf-rt-security-saml.3.3/build.gradle]: I think I already updated these

File: dev/com.ibm.ws.org.apache.cxf-rt-security.3.3/bnd.bnd [hunk 16]
```diff
@@ -0,0 +1,20 @@
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.cxf-rt-security.3.3/bnd.bnd]: done

File: dev/com.ibm.ws.org.apache.wss4j-policy.2.3.0/bnd.overrides (hunk 12 above)
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.wss4j-policy.2.3.0/bnd.overrides]: removed these for now.

File: dev/com.ibm.ws.org.apache.cxf-rt-ws-security.3.3/.classpath [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+			<accessrule kind="accessible" pattern="org/apache/cxf/binding/soap/jms/interceptor/*"/>
+			<accessrule kind="accessible" pattern="org/apache/cxf/binding/soap/model/*"/>
+			<accessrule kind="accessible" pattern="org/apache/cxf/binding/soap/saaj/*"/>
+			<accessrule kind="accessible" pattern="org/apache/cxf/binding/soap/spring/*"/>
+			<accessrule kind="accessible" pattern="org/apache/cxf/binding/soap/wsdl/extensions/*"/>
+			<accessrule kind="accessible" pattern="org/apache/cxf/binding/soap/wsdl11/*"/>
+			<accessrule ignoreifbetter="true" kind="discouraged" pattern="**"/>
+		</accessrules>
+	</classpathentry>
+	<classpathentry kind="lib" path="/com.ibm.ws.org.apache.cxf-rt-security.3.3/build/libs/com.ibm.ws.org.apache.cxf-rt-security.3.3.6.jar"/>
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.cxf-rt-ws-security.3.3/.classpath]: right. updated it

File: dev/com.ibm.ws.org.apache.cxf-rt-ws-security.3.3/src/org/apache/cxf/ws/security/cache/CXFEHCacheReplayCache.java [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+import org.apache.cxf.buslifecycle.BusLifeCycleListener;
+import org.apache.cxf.buslifecycle.BusLifeCycleManager;
+import org.apache.wss4j.common.cache.EHCacheReplayCache;
+import org.apache.wss4j.common.ext.WSSecurityException;
+
+/**
+ * Wrap the default WSS4J EHCacheReplayCache in a BusLifeCycleListener, to make sure that
+ * the cache is shutdown correctly.
+ */
+public class CXFEHCacheReplayCache extends EHCacheReplayCache implements BusLifeCycleListener {
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.cxf-rt-ws-security.3.3/src/org/apache/cxf/ws/security/cache/CXFEHCacheReplayCache.java]: nothing changed. I just had to recompile this class

File: dev/com.ibm.ws.org.apache.neethi/bnd.overrides [hunk 19]
```diff
@@ -1,3 +1,13 @@
+#*******************************************************************************
+# Copyright (c) 2019 IBM Corporation and others.
```
[the remark this answers is not in the document]
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.neethi/bnd.overrides:2]: I don't think this is new.

File: dev/com.ibm.ws.org.apache.wss4j-ws-security-stax.2.3.0/bnd.overrides (hunk 6 above)
Comment (arunavemulapalli) [reply dev/com.ibm.ws.org.apache.wss4j-ws-security-stax.2.3.0/bnd.overrides]: removed these for now

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
#build

File: dev/com.ibm.ws.org.opensaml.opensaml.messaging.impl.3.4.5/bnd.bnd [hunk 20]
```diff
@@ -0,0 +1,17 @@
...
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= jar:${fileuri;${repo;org.opensaml:opensaml-messaging-impl;3.4.5}}!/META-INF/MANIFEST.MF,bnd.overrides
+
+-buildpath: \
+    org.opensaml:opensaml-messaging-impl;version=3.4.5,\
+    com.ibm.ws.org.opensaml.opensaml-messaging.api.3.4.5;version=latest, \
```
Comment (neuwerk) [inline dev/com.ibm.ws.org.opensaml.opensaml.messaging.impl.3.4.5/bnd.bnd:15]:
```suggestion
    com.ibm.ws.org.opensaml.opensaml.messaging.api.3.4.5;version=latest, \
```

**Review by neuwerk — APPROVED** [review]
Looks good, so long as the PB passes and you update that one buildpath entry.

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
#run-libby-bot

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
#build

**Comment by arunavemulapalli** [comment]
latest pb looks good except one unrelated test failure. merging.

**Comment by arunavemulapalli** [comment]
fixed #13395

**Pull request merged**